src/Language/PureScript/TypeChecker/Kinds.hs

```diff
@@ -666,7 +695,7 @@ inferTypeSynonym moduleName (ann, tyName, tyArgs, tyBody) = do
   let (sigBinders, tyKind') = fromJust . completeBinderList $ tyKind
   bindLocalTypeVariables moduleName (first ProperName . snd <$> sigBinders) $ do
     kindRes <- freshKind (fst ann)
-    tyArgs' <- for tyArgs . traverse . maybe (freshKind (fst ann)) $ replaceAllTypeSynonyms <=< apply <=< flip checkKind E.kindType
+    tyArgs' <- for tyArgs . traverse . maybe (freshKind (fst ann)) $ replaceAllTypeSynonyms <=< apply <=< checkIsSaturatedType
```

Comment:
Does changing this in the `inferTypeSynonym` case prevent any of the current dodgy-but-not-really uses of synonyms, where you can do things like:

```
type Things f =
  { thing1 :: f Unit Number
  , thing2 :: f String Boolean
  }

type Lefts :: Type -> Type -> Type
type Lefts l r = l

type LeftThings = Things Lefts
```
?

Reply:
It shouldn't; this line change is about the kinds of type synonym parameters. So if you had
```purs
type Foo (a :: Some Type Stuff) = ...
```
and any synonyms in `Some Type Stuff` didn't end up expanding, this change may affect the error displayed (change it from KindsDoNotUnify to PartiallyAppliedSynonym).

The test for this particular line change, which I just re-verified fails if I revert it, is PASTrumpsKDNU6.

Reply:
Ah right, yep!